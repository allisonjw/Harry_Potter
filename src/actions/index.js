export const addCharacters = characters => ({
    type: 'ADD_CHARACTERS',
    characters
});

export const addHouses = houses => ({
    type: 'ADD_HOUSES',
    houses
});

export const addSpells = spells => ({
    type: 'ADD_SPELLS',
    spells
});

export const isLoading = isLoading => ({
    type: 'SET_LOADING',
    isLoading
});

export const addHouseCharacters = filteredStudents => ({
    type: 'SET_CHARACTERS',
    filteredStudents
});

export const addFavorite = (id) => ({
    type: 'ADD_FAVORITE',
    id
});
  
export const deleteFavorite = (id) => ({
    type: 'DELETE_FAVORITE',
    id
});