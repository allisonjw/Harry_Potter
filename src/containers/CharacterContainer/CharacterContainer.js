import React from 'react';
import './CharacterContainer.scss';
import CharacterCard from '../../components/CharacterCard/CharacterCard';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { PropTypes } from 'prop-types';


export const CharacterContainer = props => {
  const { filteredCharacters } = props

    const displayCharacters = filteredCharacters.map(character => {
          return (
            <CharacterCard
            {...character}
            key={character._id}
            />
          );
    });

    return (
      <section className="section_character-main">
        <section className="section_character-heading">
          <Link to='/houses'>
            <button className='back_btn' >Back to Houses</button>
          </Link>
          <h2 className='character_heading'>Characters</h2>
          </section>
      <section className="section_characters">
        {displayCharacters}
        </section>
      </section>
    )
};

export const mapStateToProps = state => ({
  filteredCharacters: state.filteredCharacters,
});

export default connect(mapStateToProps)(CharacterContainer);

CharacterContainer.propTypes = {
  filteredCharacters: PropTypes.array,
}

