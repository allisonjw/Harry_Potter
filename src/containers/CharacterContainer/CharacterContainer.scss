.section_houses-characters {
    background-image: url('https://i.ytimg.com/vi/FWZsya9giqA/maxresdefault.jpg');
    background-repeat: no-repeat;
    background-position: center;
    background-size: cover;
    height: 100vh;
    display: grid;
    grid-template-columns: repeat(auto-fit, minmax(500px, 2fr));    grid-template-rows: 1fr 1fr;
    grid-template-areas: 
    "card" "card"
    "card" "card";
    overflow-x: scroll;
}