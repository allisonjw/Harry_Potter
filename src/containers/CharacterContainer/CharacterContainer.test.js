import React from 'react';
import { shallow } from 'enzyme';
import { addHouseCharacters } from '../../actions/index'
import { CharacterContainer, mapStateToProps } from '../CharacterContainer/CharacterContainer';

describe('CharacterContainer', () => {

  describe('FilteredCharacters', () => {
        const mockFilteredCharacters = [
            {
                _id: "5a0fa54aae5bc100213c232f",
                name: "Bathsheda Babbling",
                role: "Professor, Ancient Runes",
                school: "Hogwarts School of Witchcraft and Wizardry",
                __v: 0,
                ministryOfMagic: false,
                orderOfThePhoenix: false,
                dumbledoresArmy: true,
                deathEater: false,
                bloodStatus: "unknown",
                species: "human"
            },
            {
                _id: "5a0fa5deae5bc100213c2330",
                name: "Ludo Bagman",
                role: "Head, Department of Magical Games and Sports",
                __v: 0,
                ministryOfMagic: true,
                orderOfThePhoenix: false,
                dumbledoresArmy: false,
                deathEater: false,
                bloodStatus: "unknown",
                species: "human"
            }
          ];

          const mockAddHouseCharacters = [
            { 
              _id: "5a0fa4daae5bc100213c232e",
              name: "Hannah Abbott",
              role: "student",
              house: "Hufflepuff",
              school: "Hogwarts School of Witchcraft and Wizardry",
              __v: 0,
              ministryOfMagic: false,
              orderOfThePhoenix: false,
              dumbledoresArmy: true,
              deathEater: false,
              bloodStatus: "half-blood",
              species: "human"
            }
          ];

    const wrapper = shallow(<CharacterContainer  
        filteredCharacters={mockFilteredCharacters}
        addHouseCharacters={mockAddHouseCharacters}/>

    );

    it('should match snapshot with all information passing in correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });
  });
    describe('mapStateToProps', () => {
        it('should return all state with their properties', () => {
            const mockState = {
                "filteredCharacters": [   
                  { 
                    _id: "5a0fa4daae5bc100213c232e",
                    name: "Hannah Abbott",
                    role: "student",
                    house: "Hufflepuff",
                    school: "Hogwarts School of Witchcraft and Wizardry",
                    __v: 0,
                    ministryOfMagic: false,
                    orderOfThePhoenix: false,
                    dumbledoresArmy: true,
                    deathEater: false,
                    bloodStatus: "half-blood",
                    species: "human"
                   }
                 ]
             };

             const expected = {
                "filteredCharacters": [
                   { 
                    _id: "5a0fa4daae5bc100213c232e",
                    name: "Hannah Abbott",
                    role: "student",
                    house: "Hufflepuff",
                    school: "Hogwarts School of Witchcraft and Wizardry",
                    __v: 0,
                    ministryOfMagic: false,
                    orderOfThePhoenix: false,
                    dumbledoresArmy: true,
                    deathEater: false,
                    bloodStatus: "half-blood",
                    species: "human"
                  }
                ]
             };
             const mappedProps = mapStateToProps(mockState);
             expect(mappedProps).toEqual(expected);
        });
    });
});