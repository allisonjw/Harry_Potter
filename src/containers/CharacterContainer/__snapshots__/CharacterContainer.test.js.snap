// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterContainer FilteredCharacters should match snapshot with all information passing in correctly 1`] = `
<section
  className="section_character-main"
>
  <section
    className="section_character-heading"
  >
    <Link
      to="/houses"
    >
      <button
        className="back_btn"
      >
        Back to Houses
      </button>
    </Link>
    <h2
      className="character_heading"
    >
      Characters
    </h2>
  </section>
  <section
    className="section_characters"
  >
    <Connect(CharacterCard)
      __v={0}
      _id="5a0fa54aae5bc100213c232f"
      bloodStatus="unknown"
      deathEater={false}
      dumbledoresArmy={true}
      key="5a0fa54aae5bc100213c232f"
      ministryOfMagic={false}
      name="Bathsheda Babbling"
      orderOfThePhoenix={false}
      role="Professor, Ancient Runes"
      school="Hogwarts School of Witchcraft and Wizardry"
      species="human"
    />
    <Connect(CharacterCard)
      __v={0}
      _id="5a0fa5deae5bc100213c2330"
      bloodStatus="unknown"
      deathEater={false}
      dumbledoresArmy={false}
      key="5a0fa5deae5bc100213c2330"
      ministryOfMagic={true}
      name="Ludo Bagman"
      orderOfThePhoenix={false}
      role="Head, Department of Magical Games and Sports"
      species="human"
    />
  </section>
</section>
`;
